### PR TITLE
Support for Kibana 7.0.1

### PR DESCRIPTION
**Pull request by mihirsoni**
*Issue #, if available:* #43 

**Description of changes:**
This PR enable support for Kibana 7.0.1
* Breaking changes for [Kibana](https://www.elastic.co/guide/en/kibana/current/breaking-changes-7.0.html)
* Breaking changes for [Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html)
* Backend plugin dependent  PR opendistro-for-elasticsearch/alerting#41

- [x] ~Updated configs~
- [x] ~Updated `aggregation` to use `.value`~
- [x] ~Update Update Monitor / Destinations with seq_no and primary_term~
- [x] ~Updated breaking test cases~
- [x] ~Support Dark mode  (This needs to be read form the Global settings~

dark-alerting

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mihirsoni** [comment]
test this

File: public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.js [hunk 1]
```diff
@@ -54,8 +48,8 @@ export function getTypeFromMappings(mappings, dataTypes, path = '') {
 
 export function getPathsPerDataType(mappings) {
   const dataTypes = {};
-  Object.entries(mappings)
-    .map(getDocMappings)
-    .forEach(docMappings => getTypeFromMappings(docMappings, dataTypes));
+  Object.entries(mappings).forEach(([index, { mappings: docMappings }]) =>
```
Comment (dbbaughe) [inline public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.js]: Where is docMappings from?

Comment (dbbaughe) [reply public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.js]: Whoops, misread, just renaming mappings to docMappings

File: public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.test.js [hunk 2]
```diff
@@ -20,18 +20,16 @@ describe('getPathsPerDataType', () => {
...
-              '@timestamp': { type: 'date' },
-              username: { type: 'keyword' },
-              memory: { type: 'double' },
-              phpmemory: { type: 'long' },
-              bytes: { type: 'long' },
-              clientip: { type: 'ip' },
-              id: { type: 'integer' },
-              ip: { type: 'ip' },
-            },
+          properties: {
```
Comment (dbbaughe) [inline public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.test.js]: Was the removal of types (_doc) a breaking change or a deprecation in 7.0? If it's just a deprecation we'd have to expect both.

File: public/pages/Destinations/containers/CreateDestination/CreateDestination.js [hunk 3]
```diff
@@ -49,20 +49,25 @@ class CreateDestination extends React.Component {
...
     const { location, edit, history } = this.props;
     let destinationVersion;
+    let ifSeqNo, ifPrimaryTerm;
     if (edit) {
       // In case user is refreshing in edit mode , redirect them to the destination page.
       // TODO:: Ideally this should fetch the destination from ElasticSearch and fill in value
       const destinationToEdit = _.get(location, 'state.destinationToEdit', null);
       if (destinationToEdit) {
         initialValues = { ...destinationToFormik(destinationToEdit) };
         destinationVersion = destinationToEdit.version;
```
Comment (dbbaughe) [inline public/pages/Destinations/containers/CreateDestination/CreateDestination.js]: Do we need this version anymore?

File: public/pages/MonitorDetails/containers/MonitorDetails.js [hunk 4]
```diff
@@ -91,12 +103,20 @@ export default class MonitorDetails extends Component {
...
+        params: { monitorId },
+      },
+      httpClient,
+    } = this.props;
+    const { monitor, monitorVersion, ifSeqNo, ifPrimaryTerm } = this.state;
     this.setState({ updating: true });
     return httpClient
-      .put(`../api/alerting/monitors/${id}?version=${version}`, { ...monitor, ...update })
+      .put(
+        `../api/alerting/monitors/${monitorId}?version=${monitorVersion}&ifSeqNo=${ifSeqNo}&ifPrimaryTerm=${ifPrimaryTerm}`,
```
Comment (dbbaughe) [inline public/pages/MonitorDetails/containers/MonitorDetails.js]: Do we still need monitorVersion here?

File: server/services/DestinationsService.js [hunk 5]
```diff
@@ -36,11 +36,12 @@ export default class DestinationsService {
   updateDestination = async (req, h) => {
     try {
       const { destinationId } = req.params;
-      const { version } = req.query;
+      const { version, ifSeqNo, ifPrimaryTerm } = req.query;
```
Comment (dbbaughe) [inline server/services/DestinationsService.js]: Remove version if we're not using it anymore

File: public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.test.js (hunk 2 above)
Comment (mihirsoni) [reply public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.test.js]: in 7.0 it is deprecated. but this use case is around using [get mapping](https://www.elastic.co/guide/en/elasticsearch/reference/7.0/indices-get-mapping.html) which will always return in the format we're expecting unless explicitly setting value.

File: public/pages/Destinations/containers/CreateDestination/CreateDestination.js (hunk 3 above)
Comment (mihirsoni) [reply public/pages/Destinations/containers/CreateDestination/CreateDestination.js]: Removed unused version from both the places.

**Pull request merged**